src/shared/test/integration/inter-frame-communication-test.js

```diff
@@ -132,8 +130,7 @@ describe('PortProvider-PortFinder-Bridge integration', () => {
     });
 
     // host frame
-    const portProvider = new PortProvider(window.location.origin);
-    portProvider.listen();
+    new PortProvider(window.location.origin);
```

Comment:
When I initially read this I thought there was a leak because the `PortProvider` instance was not cleaned up. Looking above I see that `PortProvider` is not in fact the original `PortProvider` class but a subclass. I think it would be clearer what was going on if instead of subclassing `PortProvider` you instead used a helper function to populate the `destroyables` list.

Reply:
Yes, I thought about adding a helper function, however, I thought it was going to be a little bit repetitive (`createBridge`, `createPortProvider`, `createPortFinder`). If implemented more generically, like `addToDestroyables(new Bridge())`, there is the possibility that one can forget the wrapping utility and we leak resources. 

I thought the subclassing was a middle ground, avoid leaking resources and not verbose, but I understand it is less intuitive.

I also thought about overriding the `new` operator, but that's not possible.

Any way, I am happy to swap subclassing for the helper utility if you want. https://github.com/hypothesis/client/pull/4090 would be a better PR to discuss that.